dataset/aoi.py

```diff
@@ -303,9 +303,6 @@ def __init__(self, raster: Union[Path, str],
 
         self.raster_stats = self.calc_raster_stats() if raster_stats else None
 
-        self.raster_np = self.raster.read()
-        self.raster.close()
```

Comment:
This could have worked, but we want to skip reading the raster in memory if possible, since we often deal with raster over 5-10 Gb (compressed). I moved to close() operation to the tests only.